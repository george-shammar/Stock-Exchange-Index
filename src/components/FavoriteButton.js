import React from 'react';
import PropTypes from 'prop-types';
import '../Index.css';

const FavoriteButton = ({ children, onClickHandler }) => (
  <button className="favorite-button" type="button" onClick={onClickHandler}>
    {children}
  </button>
);

FavoriteButton.propTypes = {
  children: PropTypes.string,
  onClickHandler: PropTypes.func,
};
FavoriteButton.defaultProps = { children: '', onClickHandler: undefined };

export default FavoriteButton;
