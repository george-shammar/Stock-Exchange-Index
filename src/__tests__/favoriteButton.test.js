import React from 'react';
import renderer from 'react-test-renderer';
import ReactDom from 'react-dom';
import FavoriteButton from '../components/FavoriteButton';

it('renders the FavoriteButton component correctly', () => {
  const tree = renderer.create(<FavoriteButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the FavoriteButton component correctly with props', () => {
    const tree = renderer.create(<FavoriteButton onClickHandler={() => console.log("stock")}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders FavoriteButton without crashing', () => {
  const div = document.createElement('div');

  ReactDom.render(
      <FavoriteButton />, 
  div);

  ReactDom.unmountComponentAtNode(div);
});