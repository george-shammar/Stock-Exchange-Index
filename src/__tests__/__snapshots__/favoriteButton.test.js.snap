// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FavoriteButton component correctly 1`] = `
<button
  className="favorite-button"
  type="button"
>
  
</button>
`;

exports[`renders the FavoriteButton component correctly 2`] = `
<button
  className="favorite-button"
  onClick={[Function]}
  type="button"
>
  
</button>
`;

exports[`renders the FavoriteButton component correctly with props 1`] = `
<button
  className="favorite-button"
  onClick={[Function]}
  type="button"
>
  
</button>
`;
